packages/commons/src/auth/jwk.ts

```diff
@@ -45,17 +46,27 @@ function assertNotPrivateKey(key: string): void {
   throw notAllowedPrivateKeyException();
 }
 
+export function assertValidRSAKeyLength(
+  keyString: string,
+  keyobject: KeyObject,
+  minLength: number = 2048
+): void {
+  const length = keyobject.asymmetricKeyDetails?.modulusLength;
+  if (!length || keyobject.asymmetricKeyType !== "rsa") {
+    throw invalidKey(keyString, "Not an RSA key");
+  }
+  if (length < minLength) {
+    throw invalidKeyLength(length, minLength);
+  }
+}
+
 export function createPublicKey(key: string): KeyObject {
   const pemKey = decodeBase64ToPem(key);
-
   assertNotPrivateKey(pemKey);
   assertNotCertificate(pemKey);
-
-  try {
-    return crypto.createPublicKey(pemKey);
-  } catch (error) {
-    throw invalidKey(key, error);
-  }
+  const publicKey = crypto.createPublicKey(pemKey);
```

Comment:
If we remove this try-catch around `crypto.createPublicKey` we don't catch errors that it could throw in case the key is not a valid public key 🥲  
We can create a function that does the try-catch and return the publicKey, the call it here

```ts
function tryToCreatePublicKey(key: string): void {
  try {
    return  crypto.createPrivateKey(key);
  } catch {
      throw invalidPublicKey();
  }
}